Components/Queues/Drivers/PhpAmqpLibDriver.php

```diff
@@ -0,0 +1,357 @@
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\Context;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
+use Smartbox\Integration\FrameworkBundle\Service;
+
+/**
+ * Class PhpAmqpLibDriver.
+ */
+class PhpAmqpLibDriver extends Service implements AsyncQueueDriverInterface
+{
+    use UsesSerializer;
+
+    /**
+     * This field specifies the prefetch window size in octets.
+     * The server will send a message in advance if it is equal to or smaller in size than the available prefetch size
+     * (and also falls into other prefetch limits).
+     * May be set to zero, meaning "no specific limit", although other prefetch limits may still apply.
+     * The prefetch-size is ignored if the no-ack option is set.
+     */
+    const PREFETCH_SIZE = null;
+
+    /**
+     * Represents the amount of message to consume by iteration
+     */
+    const PREFETCH_COUNT = 10;
+
+    /*
+     * Default exchange name
+     */
+    const EXCHANGE_NAME = '';
+
+    /**
+     * @var int Time in ms
+     */
+    private $dequeueTimeMs = 0;
+
+    /**
+     * @var string
+     */
+    protected $format = self::FORMAT_JSON;
+
+    /**
+     * @var AMQPChannel|null
+     */
+    private $channel;
+
+    /**
+     * @var \AMQPExchange|null
+     */
+    private $exchange;
+
+    /**
+     * @var array|null
+     */
+    protected $connectionsData;
+
+    /**
+     * @var array
+     */
+    protected $amqpConnections = [];
+
+    /**
+     * Number of the port used by the broker
+     *
+     * @var int $port
+     */
+    protected $port;
+
+    /**
+     * Method responsible to catch the parameters to configure the driver.
+     *
+     * {@inheritDoc}
+     */
+    public function configure(string $host, string $username, string $password, string $vhost = null)
+    {
+        $this->connectionsData[] = [
+            'host' => $host,
+            'user' => $username,
+            'password' => $password,
+            'vhost' => $vhost,
+            'port' => $this->port
+        ];
+    }
+
+    /**
+     * Method responsible to open the connection with the broker.
+     *
+     * @param bool $shuffle - used to randomize the connections if exists more than one
+     *
+     * @throws \Exception
+     */
+    public function connect($shuffle = true)
+    {
+        $this->validateConnectionData();
+        if ($shuffle) {
+            $this->shuffleConnections();
+        }
+
+        $this->addConnection(AMQPStreamConnection::create_connection($this->connectionsData, [
+            'insist' => true,
+            'login_method' => 'AMQPLAIN',
+            'locale' => 'en_UK',
+            'connection_timeout' => 60.0,
+            'read_write_timeout' => 50.0,
+            'context' => null,
+            'keepalive' => true,
+            'heartbeat' => 0,
```

Comment:
Are all this parameters required? and how did you determine these values?

Reply:
These parameters are just for test and should be configurable as soon as possible. in a soon future we can decide a way to define these, for now I think is better to leave it in the code to bring some "easy mode" to do it later.

Reply:
if these parameters are for test, does it mean that before we put this in production, these parameters need to be reviewed? What happens if we leave the default values?

Reply:
The driver gets the default values.It means, different locale, timeouts and heartbeats.But, nothing too different.

Reply:
Most likely we will put this config in the parameters of the bundle. I'd like to have the heartbeat enabled and a more reasonable timeout, but only for prod. If you set a strict timeout, debugging becomes impossible (Driver throws an exception if it misses a heartbeat)

Reply:
Based on the last Bruno change, now there is no hartbeat and neither timeout, as you mentioned @andreskrey if you are interested to have the heartbeat, we should add it in the driver configuration

Reply:
There's a new branch with the new parameters. We have to discuss more how to pass it to the application, but internally is working.

Reply:
There is a new branch with the parameters, params/amqp.